src/configs/base_keyspecs.c4m

```diff
@@ -1765,6 +1765,153 @@ Note this only specifies the base image of the final build target.
+  via `FROM <alias>` the root base image is going to be reported
+
+```
+FROM alpine as one
+FROM ubuntu:latest as two
+FROM busybox@sha256:9ae97d36d26566ff84e8893c64a6dc4fe8ca6d1144bf5b87b2b85a32def253c7 as three
+FROM nginx:1.27.0@sha256:97b83c73d3165f2deb95e02459a6e905f092260cd991f4c4eae2f192ddb99cbe as four
+FROM one as five
+FROM two
+```
+
+`DOCKER_BASE_IMAGES` is going to be:
+
+```
+{
+  "one": {
+    "name": "alpine",
+    "repo": "alpine"
+  },
+  "two": {
+    "name": "ubuntu:latest",
+    "repo": "ubuntu",
+    "tag": "latest"
+  },
+  "three": {
+    "name": "busybox@sha256:9ae97d36d26566ff84e8893c64a6dc4fe8ca6d1144bf5b87b2b85a32def253c7",
+    "repo": "busybox",
+    "digest": "9ae97d36d26566ff84e8893c64a6dc4fe8ca6d1144bf5b87b2b85a32def253c7"
+  },
+  "four": {
+    "name": "nginx:1.27.0@sha256:97b83c73d3165f2deb95e02459a6e905f092260cd991f4c4eae2f192ddb99cbe",
+    "repo": "nginx",
+    "tag": "1.27.0",
+    "digest": "97b83c73d3165f2deb95e02459a6e905f092260cd991f4c4eae2f192ddb99cbe"
+  }
+  "five": {
+    "name": "alpine",
+    "repo": "alpine"
+  },
+  "": {
+    "name": "ubuntu:latest",
+    "repo": "ubuntu",
+    "tag": "latest"
+  }
+}
+```
+"""
+}
+
+keyspec DOCKER_COPY_IMAGES {
+    kind:             ChalkTimeArtifact
+    type:             dict[string, list[dict[string, string]]
+    standard:         true
+    system:           false
+    since:            "0.4.9"
+    doc: """
+All images used in the Dockerfile `COPY --from` directives for all its sections.
+If the `COPY --from` is made from another target in the Dockerfile, it is not
+reported by this key.
+All `COPY` statements are grouped by the Dockerfile target/section.
+
+* from - name as `COPY` references image. Can be either base image or Dockerfile target/section name
+* name - full name as its referenced in Dockerfile
+* repo - only the repo of the image
+* tag - only the tag of the image (if `:<tag>` is provided)
+* digest - only the digest of the image (if `@sha256:<digest>` is provided)
+* src - what COPY copies from image
+* dest - where COPY copies to in the target/section
+
+Special cases:
+
+* If the `COPY` references another target/section within the Dockerfile, its
+  base image is reported.
+```
+FROM alpine as one
+FROM ubuntu as two
+COPY --from=docker /bin/docker /bin/docker
+COPY --from=busybox:latest /bin/busybox /bin/busybox
+FROM scratch
+COPY --nginx:1.27.0@sha256:97b83c73d3165f2deb95e02459a6e905f092260cd991f4c4eae2f192ddb99cbe /bin/nginx /bin/nginx
+COPY --from=one /bin/sh /sh
+```
+
+`DOCKER_BASE_IMAGES` is going to be:
+
+```
+{
+  "two": [
+    {
+      "from": "docker",
+      "name": "docker",
+      "repo": "docker",
+      "src": ["/bin/docker"],
+      "dest": "/bin/docker",
+    },
+    {
+      "from": "busybox:latest",
+      "name": "busybox:latest",
+      "repo": "busybox",
+      "tag": "latest",
+      "src": ["/bin/busybox"],
+      "dest": "bin/busybox",
+    }
+  ],
```

Comment:
Shouldn't this also have arefence two `ubuntu` as part of two as well?

Reply:
this is for `CHALK_COPY_IMAGES` which reports information about `COPY` directives within each target/section

Reply:
Ah the text above (https://github.com/crashappsec/chalk/pull/382/files#diff-b93822976ff3c1b7f18f425dc101793754cd64424f682b5599991d37dbf2e0aeR1871) says ``DOCKER_BASE_IMAGES` is going to be:`

Reply:
copy-paste strikes again. fixed some typos. just force pushed fix

thanks for noticing 😄